src/main/kotlin/no/nav/tilleggsstonader/sak/vilkår/domain/Vilkårperiode.kt

```diff
@@ -35,6 +35,21 @@ data class Vilkårperiode(
         feilHvis(ugyldigTypeOgDetaljer) {
             "Ugyldig kombinasjon type=${type.javaClass.simpleName} detaljer=${detaljer.javaClass.simpleName}"
         }
+
+        validerSlettet()
+    }
+
+    private fun validerSlettet() {
```

Comment:
Kan denne endre så den ikke er i fortid? For denne kjøres vell før den evt. slettes? F.eks. `validerKanSlettes`? Og evt. bare kjøre funksjonen dersom resultatet er slettet? Da slipper man å ha `&& resultat = slettet` i alle sjekkene i funksjonen

Reply:
`if(slettet) validerKanSlettes()`

Reply:
Eneste er at man ikke får tatt med slettkommentar i denne funksjonen om man løser det sånn her. Kan evt gjøre noe sånt

```kotlin
       fun validerSlettefelter() {
            if(resultat == ResultatVilkårperiode.SLETTET) {
                feilHvis(kilde != KildeVilkårsperiode.MANUELL) {
                    "Kan ikke slette når kilde=$kilde"
                }
                feilHvis(slettetKommentar.isNullOrBlank()) {
                    "Mangler kommentar for resultat=$resultat"
                }
            }
            else {
                feilHvis(!slettetKommentar.isNullOrBlank()) {
                    "Kan ikke ha slettetkommentar med resultat=$resultat"
                }
            }
        }
```